plugins/woocommerce/src/Internal/Admin/Homescreen.php

```diff
@@ -49,7 +49,86 @@ public function __construct() {
 			// priority is 20 to run after https://github.com/woocommerce/woocommerce/blob/a55ae325306fc2179149ba9b97e66f32f84fdd9c/includes/admin/class-wc-admin-menus.php#L165.
 			add_action( 'admin_head', array( $this, 'update_link_structure' ), 20 );
 		}
+
 		add_filter( 'woocommerce_admin_preload_options', array( $this, 'preload_options' ) );
+
+		if ( Features::is_enabled( 'shipping-smart-defaults' ) ) {
+			add_filter(
+				'woocommerce_admin_shared_settings',
+				array( $this, 'maybe_set_default_shipping_options_on_home' ),
+				9999
+			);
+		}
+	}
+
+	/**
+	 * Set free shipping in the same country as the store default
+	 * Flag rate in all other countries when any of the following conditions are ture
+	 *
+	 * - The store sells physical products, has JP and WCS installed and connected, and is located in the US.
+	 * - The store sells physical products, and is not located in US/Canada/Australia/UK (irrelevant if JP is installed or not).
+	 * - The store sells physical products and is located in US, but JP and WCS are not installed.
+	 *
+	 * @param array $settings shared admin settings.
+	 * @return array
+	 */
+	public function maybe_set_default_shipping_options_on_home( $settings ) {
+		$current_screen = get_current_screen();
+
+		// Abort if it's not the homescreen.
+		if ( ! isset( $current_screen->id ) || 'woocommerce_page_wc-admin' !== $current_screen->id ) {
+			return $settings;
+		}
+
+		// Abort if we already created the shipping options.
+		$already_created = get_option( 'woocommerce_admin_created_default_shipping_zones' );
+		if ( 'yes' === $already_created ) {
+			return $settings;
+		}
+
+		$user_skipped_obw = $settings['onboarding']['profile']['skipped'] ?? false;
+		$store_address = $settings['preloadSettings']['general']['store_address'] ?? '';
```

Comment:
@moon0326 It seems that `store_address` is always empty, and OBW uses `woocommerce_store_address`. 🤔

Reply:
@chihsuan Thank you for catching it! I meant to use woocommerce_store_address. Updated in https://github.com/woocommerce/woocommerce/pull/33547/commits/67c8a937993f8c79b5c3df0eb32231050ed52260